### PR TITLE
feat:  validate tsconfig.json 

### PR DESCRIPTION
**Pull request by suxin2017**
## Related issue (if exists)

#2489 

### <samp>🤖 Generated by Copilot at edbc9c2</samp>

Added TypeScript configuration validation to `rspack`. The `rspack` package now depends on `get-tsconfig` and uses it to load and check the TypeScript settings for each option before compiling. The `pnpm-lock.yaml` file was updated accordingly.

<details open=true>
  <summary><h2>Walkthrough</h2></summary>

### <samp>🤖 Generated by Copilot at edbc9c2</samp>

*  Add `revalidateTsconfig` function to `rspack.ts` to load and validate TypeScript configuration files for each rspack option ([link](https://github.com/web-infra-dev/rspack/pull/2899/files?diff=unified&w=0#diff-f8cb4d24a4c7e2bccefde48c7cc6285ac03900bac4879f0b6cd40ead5608e393R115-R140))
*  Call `revalidateTsconfig` function in `rspack` function before creating compiler or multi-compiler ([link](https://github.com/web-infra-dev/rspack/pull/2899/files?diff=unified&w=0#diff-f8cb4d24a4c7e2bccefde48c7cc6285ac03900bac4879f0b6cd40ead5608e393R161))
*  Import `getTsconfig` module in `rspack.ts` to use in `revalidateTsconfig` function ([link](https://github.com/web-infra-dev/rspack/pull/2899/files?diff=unified&w=0#diff-f8cb4d24a4c7e2bccefde48c7cc6285ac03900bac4879f0b6cd40ead5608e393L10-R26))
*  Reorder and remove unused imports in `rspack.ts` for clarity and consistency ([link](https://github.com/web-infra-dev/rspack/pull/2899/files?diff=unified&w=0#diff-f8cb4d24a4c7e2bccefde48c7cc6285ac03900bac4879f0b6cd40ead5608e393L28-R35))
*  Add `get-tsconfig` dependency to `pnpm-lock.yaml` under `rspack` package and `packages` section ([link](https://github.com/web-infra-dev/rspack/pull/2899/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR628), [link](https://github.com/web-infra-dev/rspack/pull/2899/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR660), [link](https://github.com/web-infra-dev/rspack/pull/2899/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR10107-R10110))

</details>

### CODE REVIEW

File: packages/rspack/src/rspack.ts [hunk 1]
```diff
@@ -111,6 +112,32 @@ function revalidateWithStrategy(options: RspackOptions | MultiRspackOptions) {
 	}
 }
 
+function revalidateTsconfig(options: RspackOptions | MultiRspackOptions) {
```
Comment (hyf0) [inline packages/rspack/src/rspack.ts:115]: IMO, we should only detect `tsconfig.json` while using CLI. I'm not sure whether should have this logic in `@rspack/core`.

@hardfist cc

There is a similar feature at https://github.com/web-infra-dev/rspack/pull/2575

**Review by hyf0 — APPROVED** [review]

**Comment by hyf0** [comment]
I also wonder: Does the value of `tsconfig.json` actually affect the behaviors of Rspack? @hardfist cc

File: packages/rspack/src/rspack.ts (hunk 1 above)
Comment (suxin2017) [reply packages/rspack/src/rspack.ts:115]: So do I need to put my code in the cli?

**Comment by suxin2017** [comment]
> I also wonder: Does the value of `tsconfig.json` actually affect the behaviors of Rspack? @hardfist cc

I don't think this will affect rspack but only swc

File: packages/rspack/src/rspack.ts (hunk 1 above)
Comment (hyf0) [reply packages/rspack/src/rspack.ts:115]: @hardfist cc

Comment (hardfist) [reply packages/rspack/src/rspack.ts:115]: we shouldn't put this logic into core, @rspack/core should align with webpack

Comment (suxin2017) [reply packages/rspack/src/rspack.ts]: I refactored the code and put it in @rspack/cli cc @hardfist

**Pull request closed without merging**